lib/errors/errors.go

```diff
@@ -65,23 +65,33 @@ type ErrorValidation struct {
 }
 
 type ErroredField struct {
-	Name  string
-	Value interface{}
+	Name   string
+	Value  interface{}
+	Reason string
 }
 
 func (e ErrorValidation) Error() string {
 	if len(e.ErroredFields) == 0 {
 		return "unknown validation error"
 	} else if len(e.ErroredFields) == 1 {
-		return fmt.Sprintf("error with field %s = '%v'",
-			e.ErroredFields[0].Name,
-			e.ErroredFields[0].Value)
+		f := e.ErroredFields[0]
+		if f.Reason == "" {
+			return fmt.Sprintf("error with field %s = '%v'",
+				f.Name, f.Value)
```

Comment:
Good question,  it is possible that the Value could be nil I guess (if someone specified null in the json)

But Sprintf handles it gracefully (it writes out `<nil>`)